### ISSUE TITLE
Improve error handling for malformed components

### ISSUE THREAD

**Issue by spiffytech** [forgojs/forgo#13]
The type `forgo.ForgoComponent` does not cause a compile error if the component function returns a node:

```tsx
function ShouldNotCompile(): forgo.ForgoComponent<forgo.ForgoElementProps> {
  return <div>Oh noooooooo</div>
}
```

^^ This compiles successfully, but at runtime Fargo throws an error because it can't find `component.render`.

I suggest the following:

1. Update the error message for a missing `render` method to include the name of the component function (`fn.name`) that returned the malformed component. That would have saved me a bunch of trouble tracking down which code was the problem.
2. Update the TS types to reject that return value if Forgo isn't going to support it.

**Comment by jeswin** [comment]
> Update the error message for a missing render method to include the name of the component function (fn.name) ...

This is an excellent suggestion - not it throws an error message mentioning fn.name. Implemented in 0.0.28.

> Update the TS types to reject that return value if Forgo isn't going to support it.

I need to figure out how to do this. Will keep this open for now.

**Comment by spiffytech** [comment]
The new component API emits a TypeScript error if the return value isn't an instance of the `Component` class.
